includes/class-taxjar-customer-record.php

```diff
@@ -261,6 +233,47 @@ public function get_data_from_object() {
 		return $customer_data;
 	}
 
+	/**
+	 * Retrieves the name of the customer.
+	 * Falls back to username if no shipping, billing or account names are available.
+	 *
+	 * @return string - Customer's name
+	 */
+	public function get_customer_name() {
+		$first_name = $this->object->get_shipping_first_name();
+		$last_name = $this->object->get_shipping_last_name();
+
+		if ( empty( $first_name ) ) {
+			$first_name = $this->object->get_billing_first_name();
+			if ( empty( $first_name ) ) {
+				$first_name = $this->object->get_first_name();
+			}
+		}
+
+		if ( empty( $last_name ) ) {
+			$last_name = $this->object->get_billing_last_name();
+			if ( empty( $last_name ) ) {
+				$last_name = $this->object->get_last_name();
+			}
+		}
+
+		if ( empty( $first_name ) && empty( $last_name ) ) {
+			$name = '';
+		} else if ( empty( $first_name ) ) {
+			$name = $last_name;
+		} else if ( empty( $last_name ) ) {
+			$name = $first_name;
+		} else {
+			$name = $first_name . ' ' . $last_name;
+		}
+
+		if ( empty( $name ) ) {
```

Comment:
If all the other potential name fields are empty, it now falls back to the username.